src/2d/shallow/qinit.f90

```diff
@@ -26,6 +27,10 @@ subroutine qinit(meqn,mbc,mx,my,xlower,ylower,dx,dy,q,maux,aux)
         call add_perturbation(meqn,mbc,mx,my,xlower,ylower,dx,dy,q,maux,aux)
     endif
 
+    ! add an almost trivial amount to the cells holding point sources
+    call point_sources%apply_trivial_ic(&
```

Comment:
Is this required for stability reasons?  I might suggest keeping this below `dry_tolerance` rather than slightly above.  Also this may cause an arbitrary amount of initial fluid to be added to the domain depending on the size of the containing grid cell along with some weird behavior if you have multiple resolutions under the point source.

Reply:
The reasons I added this are:
1. to trigger an initial adaptive refinement around point sources, and
2. for the solver to properly predict the time-step size of the second time-step.

For our cases, at the beginning, there is no working fluid in the domain. And I found the solver, therefore, fails to predict the proper time-step size for the second time-step in some cases. In these cases, the solver uses a very large time-step, that is likely the time gap between the current time and the time for the next result output.

I agree this workaround is not that justified. But I wasn't sure how to resolve the two abovementioned problems without touching much original GeoClaw code. And I think if the initial amounts added to the domain are close to the `dry_tolerance`, at least the error should not affect simulations.

I agree that it may be proper to keep the amount below `dry_tolerance`. I'm going to test it.

Reply:
I'm thinking about other ways to resolve the problems.

Reply:
See the update in [this comment](https://github.com/barbagroup/geoclaw/pull/8#issuecomment-430018402).